exercises/concept-wip/arys-amazing-lasagna/.docs/introduction.md

```diff
@@ -0,0 +1,258 @@
+Although some cases it is necessary to specify the type.
+The cases that requires a type to be set will be explained in later concepts.
+
+### Variables
+
+To declare a [variable][variables], you must use the `=` assignment operator.
+In crystal should variables be written in [snake_case][snake-case].
+
+A variable can be declared and then redefine later.
+
+```crystal
+number = 1
+p number # => 1
+
+number = 2
+p number # => 2
+```
+
+### Constants
+
+[Constants][constants] are declared using the same `=` assignment operator, but uses all uppercase letters or also known as SCREAMING_SNAKE_CASE.
+The difference between constants and variables is that constants can't be redefined.
+
+```crystal
+NUMBER = 1
+p NUMBER # => 1
+```
+
+Assigning a value to a constant that is already defined will result in a compile-time error.
+
+```crystal
+NUMBER = 1
+
+NUMBER = 2
+```
+
+```console
+In test.cr:3:1
+
+ 3 | NUMBER = 2
+     ^
+Error: already initialized constant NUMBER
+```
+
+### Types
+
+Crystal has a [type inference system][type], so you don't have to specify the type of a variable or method argument.
+The compiler will try to "guess" or infer the type of the variable or method argument.
+To see the type of a variable or method argument, you can use the `class` method.
+In this case the type of the variable `number` is `Int32`, Int is short for integer.
+
+```crystal
+number = 1
+p number.class # => Int32
+```
+
+Using a method that doesn't exist on the type of the variable or method argument will result in a compile-time error.
+
+```crystal
+number = 1
+number + "1"
+```
+
+```console
+In test.cr:2:7
+
+ 2 | number + "1"
+           ^
+Error: no overload matches 'Int32#+' with type String
+```
+
+### Methods
+
+[Methods][methods] are a way to group together a set of instructions that can be reused.
+In crystal are methods often defined in a class, module or struct.
+But methods can also be defined outside of a class, module or struct.
+Methods are declared using the `def` keyword, followed by the name of the method.
+In crystal should methods be written in snake_case.
+When a method doesnt have any arguments, you can omit the parentheses.
+To declare the end of a method, you must use the `end` keyword.
+
+```crystal
+def hello
+  puts "Hello World!"
+end
+```
+
+### Arguments
+
+Methods can have [arguments][arguments].
+Arguments are data that is passed to a method.
+To be able to give a method arguments, you must specify the name of the argument.
+In crystal should arguments be written in snake_case.
+A method can have multiple arguments, but the arguments must be separated by a comma.
+
+```crystal
+def hello(name)
+  puts "Hello #{name}!"
+end
+```
+
+In crystal it is possible to give a method argument a [default value][default_arguments].
+When a method argument has a default value, it is not required to pass that argument when calling the method.
+
+```crystal
+def hello(name = "World")
+  "Hello #{name}!"
+end
+```
```

Comment:
The introduction should _only_ explain things the student needs to solve the exercise (or understand the stub file).
As student won't need to know about default values, it's better to postpone that to a later exercise.